packages/beacon-node/src/sync/unknownBlock.ts

```diff
@@ -825,4 +753,231 @@ export class UnknownBlockSync {
 
     return badPendingBlocks;
   }
+
+  private getMaxDownloadAttempts(): number {
+    if (this.config.getForkSeq(this.chain.clock.currentSlot) < ForkSeq.fulu) {
+      return MAX_ATTEMPTS_PER_BLOCK;
+    }
+
+    // TODO: I consider max 20 downloads per block for a supernode is enough for devnets
+    // review this computation for public testnets or mainnet
+    return Math.min(
+      20,
+      (MAX_ATTEMPTS_PER_BLOCK * this.network.custodyConfig.sampleGroups.length) / this.config.SAMPLES_PER_SLOT
+    );
+  }
+}
+
+/**
+ * Class to track active byRoots requests and balance them across eligible peers.
+ */
+export class UnknownBlockPeerBalancer {
```

Comment:
When taking into account the custody have, we will likely need to download using the same peers for both ByRange and ByRoot.  Wondering if we should have a unified PeerBalancer that accounts for both.  Might be something to look at in the future